### DISCUSSION TITLE
OsmAnd errors incoming from OSRM

### DISCUSSION THREAD

**Discussion by ignacio-agullo** [osmandapp/OsmAnd#15749] [General]
Edit: I just found that writing the name of the OsmAnd app complete with the tilde leads the GitHub editor to show the text with lines over it, so I am calling the app 'OsmAnd plus' instead.

I have been using OpenStreetMap for over 7 years.  I have noticed many route mistakes, but on October 29, I came across a mistake so much ridiculous that I decided to report it.  And then, on November 1, the mistake grew tenfold!

It happens when you plot a car route in the country of Spain, Basque Country region, Gipuzkoa province, between the towns of Beasain and Irun.  That route obviously has to follow the N-I highway.  At some point, the A-15 highway merges into the N-I.  And then, you need to take the following exit, 447B.  A 60-kilometer route.  That is what Google Maps tells you to do.

But OsmAnd plus 4.2.6 tells you to take exit 445B instead, take a narrow road where you can barely pass between two houses, climb a mount between meadows with sheep, then climb down and join *again* the N-I highway, stay on the joining lane that is an exiting lane as well, so at the end of it you take exit 447B.  A 62-kilometer route.  This is the original mistake.  As for the map used, I searched through the directories on my Android smartphone and found the map file Spain_basque-country_europe.obf from date September 1, 2022.  Now I would add here a screenshot, but GitHub disallows me from doing it.
 
I originally though that the mistake would probably come from the map, the joining/exiting lane being mapped as separated from the rest of lanes of N-I, so I created an account for myself at openstreetmap.org and reported it in the OpenStreetMap Spanish Community, along with screenshots:
Pifia en N-I en Andoain (Guipúzcoa) - Communities / España (Spain) - OpenStreetMap Community
https://community.openstreetmap.org/t/pifia-en-n-i-en-andoain-guipuzcoa/5123/3

But they shown me there that thas was not the case, for other navigation apps that also use OpenStreetMap such as GraphHopper and Organic Maps don't have that mistake.  They told me about OSRM, and I checked it out, and sure enough, the exactly same mistake was there:
OpenStreetMap Routing with Open Source Routing Machine
https://map.project-osrm.org/?z=13&center=43.205864%2C-2.007236&loc=43.047327%2C-2.203287&loc=43.338318%2C-1.788809&hl=en&alt=0&srv=0

So it seems that OsmAnd plus 4.2.6 has mistakes incoming from OSRM.  Now, I should report this mistake directly to OSRM, but sadly its website doesn't have anything resembling to a forum or a form for reporting issues.  So, as an error found in OsmAnd plus, I am reporting it here.

But wait!  I told you the mistake grew tenfold.  I noticed there was a more recent OsmAnd~Basque Country map, one from November 1, 2022, so I updated it... and then catastrophe ensued.  The 60-kilometer route turned into a 112-kilometer one!  That is the current situation with OsmAnd plus, and I can tell you that it is obviously wrong, for I followed that route on November 2, 2022.  I would add another screenshot here, but then again, no.

There are various problems here:
Map from September 1: The original 2-kilometer detour, incoming from OSRM, seems to happen because of failing fo notice that highways N-I and A-15 merge.
Map from November 1. This crazy 52-kilometer detour happens when OsmAnd plus's 'Temporary Limits' is enabled.  But even if you disable the 'Temporary Limits', you get a barely better route... only a 38-kilometer detour.

So, to recap,
-OsmAnd plus 4.2.6 working with map Spain_basque-country_europe.obf from September 1, has a 2-kilometer detour mistake incoming from OSRM.
-OsmAnd plus 4.2.6 working with map Spain_basque-country_europe.obf from November 1, has a 62-kilometer detour mistake with 'Temporary Limits' enabled and a 38-kilometer detour mistake with 'Temporary Limits' disabled.  So, this current version of the map is catastrophic in either sense, it would be better to recover the previous one.
